website/addons/s3/views/config.py

```diff
@@ -149,10 +149,13 @@ def s3_get_node_settings(auth, node_addon, **kwargs):
 @must_have_permission('write')
 @must_not_be_registration
 def s3_bucket_list(auth, node_addon, user_addon, **kwargs):
-
-    return {
-        'buckets': get_bucket_drop_down(user_addon)
-    }
+    buckets, status_code = get_bucket_drop_down(user_addon)
+    if buckets:
+        return {
+            'buckets': buckets
+        }
+    else:
```

Comment:
If you raise an `HTTPError` in `get_bucket_drop_down` as suggested in my previous comment, this branch is unnecessary.